oclc/classify.py

```diff
@@ -32,6 +36,147 @@ class OCLC(object):
     FORMAT = "OCLC.format"
 
 
+class NameParser(VIAFNameParser):
+    """Parse the name format used by OCLC Classify.
+
+    This is like the VIAF format with the addition of optional roles
+    for authors.
+
+    Example:
+     Giles, Lionel, 1875-1958 [Writer of added commentary; Translator]
+
+    In addition, multiple authors are sometimes given in one string,
+    separated by the pipe character.
+
+    Example:
+     Leodhas, Sorche Nic, 1898-1969 | Ness, Evaline [Illustrator]
+    """
+
+    ROLES = re.compile("\[([^]]+)\]$")
+
+    # Map the roles defined in OCLC Classify to the constants
+    # defined in Contributor.
+    ROLE_MAPPING = {
+        "Author": Contributor.AUTHOR_ROLE,
+	"Translator": Contributor.TRANSLATOR_ROLE,
```

Comment:
I think you have a mixture of spaces and tabs here and it's messing up the indentation in github.